### PR TITLE
support for unlocking trophies

### PR DESCRIPTION
**Pull request by CrazyBloo**
edit: this is ready for review, when Vinicius finishes the image support for imgui i will make another pr with trophy icons

### CODE REVIEW

File: src/core/libraries/np_trophy/trophy_ui.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+            TrophyInfo currentTrophyInfo = trophyQueue[0];
+            if (Begin("Trophy Window", nullptr,
+                      ImGuiWindowFlags_NoDecoration | ImGuiWindowFlags_NoSavedSettings |
+                          ImGuiWindowFlags_NoInputs)) {
+                Text("Trophy earned!");
+                TextWrapped(currentTrophyInfo.trophyName.c_str());
+
+                End();
+            }
```
Comment (viniciuslrangel) [inline src/core/libraries/np_trophy/trophy_ui.cpp]: The End should be outside the conditional (I made the same mistake in my PR)

File: src/core/libraries/np_trophy/trophy_ui.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace Libraries::NpTrophy {
+
+struct TrophyInfo {
+    int trophyId = -1;
+    std::string trophyName;
+};
+
+class TrophyUI final : public ImGui::Layer {
+    bool first_render{false};
```
Comment (viniciuslrangel) [inline src/core/libraries/np_trophy/trophy_ui.h]: unused var

Comment (CrazyBloo) [reply src/core/libraries/np_trophy/trophy_ui.h]: ah, i assumed this was needed for imgui_layer for some reason, thank you

Comment (viniciuslrangel) [reply src/core/libraries/np_trophy/trophy_ui.h]: It's not! I used it to default select the button for gamepad navigation

File: src/core/libraries/np_trophy/np_trophy.cpp [hunk 3]
```diff
@@ -4,13 +4,20 @@
 #include <unordered_map>
 
 #include "common/logging/log.h"
+#include "common/path_util.h"
 #include "common/slot_vector.h"
 #include "core/libraries/error_codes.h"
 #include "core/libraries/libs.h"
+#include "externals/pugixml/src/pugixml.hpp"
```
Comment (abouvier) [inline src/core/libraries/np_trophy/np_trophy.cpp:11]: You should use `#include <pugixml.hpp>` for compatibility with the system library.

Comment (CrazyBloo) [reply src/core/libraries/np_trophy/np_trophy.cpp:11]: i will change it, thank you

File: src/core/libraries/np_trophy/trophy_ui.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            SetNextWindowCollapsed(false);
+            SetNextWindowPos(ImVec2(io.DisplaySize.x - 200, 50));
+            KeepNavHighlight();
+
+            TrophyInfo currentTrophyInfo = trophyQueue[0];
+            if (Begin("Trophy Window", nullptr,
+                      ImGuiWindowFlags_NoDecoration | ImGuiWindowFlags_NoSavedSettings |
+                          ImGuiWindowFlags_NoInputs)) {
+                Text("Trophy earned!");
+                TextWrapped(currentTrophyInfo.trophyName.c_str());
```
Comment (abouvier) [inline src/core/libraries/np_trophy/trophy_ui.cpp:69]: Should be `TextWrapped("%s", currentTrophyInfo.trophyName.c_str());` to compile with `-Werror=format-security`.

**Comment by Evilhelder** [comment]
After this commit, some games like Bloodborne are stuck on the trophy download screen.

[shad_log.txt](https://github.com/user-attachments/files/16966893/shad_log.txt)


image
image

**Comment by CrazyBloo** [comment]
> After this commit, some games like Bloodborne are stuck on the trophy download screen.
> 
> [shad_log.txt](https://github.com/user-attachments/files/16966893/shad_log.txt)
> 
> image image

Are you sure this is from this commit? I have seen people with this issue before. It doesn't seem like any of the functions I modified are called in this log. It seems like this issue is caused by sceNpTrophyGetTrophyInfo not being implemented. I will try to implement this function today

**Comment by Evilhelder** [comment]
> > After this commit, some games like Bloodborne are stuck on the trophy download screen.
> > [shad_log.txt](https://github.com/user-attachments/files/16966893/shad_log.txt)
> > image image
> 
> Are you sure this is from this commit? I have seen people with this issue before. It doesn't seem like any of the functions I modified are called in this log. It seems like this issue is caused by sceNpTrophyGetTrophyInfo not being implemented. I will try to implement this function today

You are correct, sorry, I tested some builds before its implementation and the error persists, but it is strange, because if I test an older build like the clean 0.2.1 the game works normally (Obviously it does not go beyond character creation, but it does not give this error with the trophies.) sorry for the inconvenience .v.

**Comment by CrazyBloo** [comment]
> You are correct, sorry, I tested some builds before its implementation and the error persists, but it is strange, because if I test an older build like the clean 0.2.1 the game works normally (Obviously it does not go beyond character creation, but it does not give this error with the trophies.) sorry for the inconvenience .v.

No worries, i will try to fix it anyway

**Pull request merged**